pre_commit/clientlib.py

```diff
@@ -37,6 +37,7 @@ def _make_argparser(filenames_help):
     cfgv.Required('entry', cfgv.check_string),
     cfgv.Required('language', cfgv.check_one_of(all_languages)),
 
+    cfgv.Optional('alias', cfgv.check_string, ''),
```

Comment:
default this to `id`, then you can remove the `or` logic below and only use `alias`.

you'll need to write a custom validator -- you can see `cfgv` for examples or `MigrateShaToRev` below